plugins/default/lib/disk_usage.py

```diff
@@ -1,35 +1,95 @@
+                        required=True, help="Warning Threshold")
+    parser.add_argument('-c', '--crit', type=int, action="store",
+                        required=True, help="Critical Threshold")
+    return parser
+
+
+def parse_df_line(line):
     mo = check_disk_output_pattern.match(line)
     if mo is None:
         return ()
-    return (mo.group(1), int(mo.group(2)), int(mo.group(3)))
+    return mo.group(3), int(mo.group(1)), int(mo.group(2))
+
+
+def parse_df_lines(lines):
+    return filter(None, map(parse_df_line, lines.splitlines()))
+
+
+def analize(pod, disks, warning_threshold, critical_threshold):
+    results = []
+    for mount, space_usage, inode_usage in disks:
+        max_pcent = max(space_usage, inode_usage)
+
+        disk_status = nagios.UNKNOWN
+        if max_pcent >= critical_threshold:
+            disk_status = nagios.CRIT
+        elif max_pcent >= warning_threshold:
+            disk_status = nagios.WARN
+        elif max_pcent < warning_threshold:
+            disk_status = nagios.OK
+
+        results.append([pod, mount, space_usage, inode_usage, disk_status])
+    return results
+
 
-def analize(pcent, ipcent, warning_threshold, critical_threshold):
-    max_pcent = max(pcent, ipcent)
-    return max_pcent >= warning_threshold, max_pcent >= critical_threshold
+def report(results):
+    counts = Counter(d[4] for d in results)
+
+    ret = max(counts)
+
+    if ret == nagios.OK:
+        print "%s: All %s volumes are under the warning threshold" % (nagios.status_code_to_label(ret), len(results))
+    elif ret == nagios.UNKNOWN:
+        print "%s: Unable to determine usage on %s volumes" % (nagios.status_code_to_label(ret), counts[nagios.UNKNOWN])
+    elif ret == nagios.WARN:
+        print "%s: There are %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.WARN])
+    else:
+        print "%s: There are %s volumes over the critical threshold and %s volumes over the warning threshold" % (nagios.status_code_to_label(ret), counts[nagios.CRIT], counts[nagios.WARN])
+
+    for pod, mount, disk_usage, inode_usage, status in results:
+        print "%s: %s:%s - bytes used: %s%%, inodes used: %s%%" % (nagios.status_code_to_label(status), pod, mount, disk_usage, inode_usage)
+
+    return ret
 
-def report():
-    raise NotImplementedError("report is not implemented yet")
 
 def main():
-    project = oc("project", "-q")
-    print analize(map(parse_df, exec_in_pods(project, get_running_pod_names(), check_disk_cmd)))
+    args = generate_parser().parse_args()
+    warn = args.warn
+    crit = args.crit
+
+    if crit < warn:
+        raise ValueError("Critical threshold lower than Warning threshold")
+
+    project = openshift.get_project()
+
+    results = []
+
+    for pod, lines in openshift.exec_in_pods(project, openshift.get_running_pod_names(project), check_disk_cmd):
+        for d in analize(pod, parse_df_lines(lines), warn, crit):
+            results.append(d)
```

Comment:
Yeah we do, when parsing the lines, we have multiple disk results lines per pod.  This is why I had asked if there was a good way to use map and pass in multiple parameters, but just using extend seems to be good enough to avoid the double looping.